src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/DampedHarmonic.cpp

```diff
@@ -0,0 +1,960 @@
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+Scalar<DataType> weight_function(
+    const tnsr::I<DataType, SpatialDim, Frame>& coords,
+    const double sigma_r) noexcept {
+  Scalar<DataType> weight{};
+  GeneralizedHarmonic::DampedHarmonicGauge_detail::weight_function(
+      make_not_null(&weight), coords, sigma_r);
+  return weight;
+}
+// @}
+
+// @{
+/*!
+ * \ingroup GeneralizedHarmonic
+ * \brief Spacetime derivatives of the spatial weight function that enters the
+ * damped harmonic gauge source function.
+ *
+ * \details Compute the derivatives :
+ * \f{align*}
+ * \partial_a W(x^i)=& \partial_a \exp(- (r/\sigma_r)^2) \\
+ *                  =& (-2 * x^i / \sigma_r^2) * exp(-(r/\sigma_r)^2)
+ * \f}
+ */
+template <size_t SpatialDim, typename Frame, typename DataType>
+void spacetime_deriv_of_weight_function(
+    const gsl::not_null<tnsr::a<DataType, SpatialDim, Frame>*> d4_weight,
+    const tnsr::I<DataType, SpatialDim, Frame>& coords,
+    const double sigma_r) noexcept {
+  if (UNLIKELY(get_size(get<0>(*d4_weight)) != get_size(get<0>(coords)))) {
+    *d4_weight = tnsr::a<DataType, SpatialDim, Frame>(get_size(get<0>(coords)));
+  }
+  const DataType pre_factor =
+      get(weight_function(coords, sigma_r)) * (-2. / pow<2>(sigma_r));
+  // time derivative of weight function is zero
+  get<0>(*d4_weight) = 0.;
+  for (size_t i = 0; i < SpatialDim; ++i) {
+    d4_weight->get(1 + i) = pre_factor * coords.get(i);
+  }
+}
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+tnsr::a<DataType, SpatialDim, Frame> spacetime_deriv_of_weight_function(
+    const tnsr::I<DataType, SpatialDim, Frame>& coords,
+    const double sigma_r) noexcept {
+  tnsr::a<DataType, SpatialDim, Frame> d4_weight{};
+  GeneralizedHarmonic::DampedHarmonicGauge_detail::
+      spacetime_deriv_of_weight_function(make_not_null(&d4_weight), coords,
+                                         sigma_r);
+  return d4_weight;
+}
+// @}
+
+/*!
+ * \ingroup GeneralizedHarmonic
+ * \brief Roll-on function for the damped harmonic gauge.
+ *
+ * \details For times after \f$t_0\f$, compute the roll-on function
+ * \f$ R(t) = 1 - \exp(-((t - t_0)/\sigma_t)^4]) \f$,
+ * and return \f$ R(t) = 0\f$ at times before.
+ */
+double roll_on_function(const double time, const double t_start,
+                        const double sigma_t) noexcept {
+  if (time < t_start) {
+    return 0.;
+  }
+  return 1. - exp(-pow<4>((time - t_start) / sigma_t));
+}
+
+/*!
+ * \ingroup GeneralizedHarmonic
+ * \brief Time derivative of the damped harmonic gauge roll-on function.
+ *
+ * \details Compute the time derivative:
+ * \f{align*}
+ * \partial_0 R(t) = 4 exp[-((t - t0)/\sigma_t)^4] (t - t0)^3 / sigma_t^4
+ * \f}
+ */
+double time_deriv_of_roll_on_function(const double time, const double t_start,
+                                      const double sigma_t) noexcept {
+  if (time < t_start) {
+    return 0.;
+  }
+  const double tnrm = (time - t_start) / sigma_t;
```

Comment:
I'm finding the variable name `tnrm` difficult to understand. Is there a better name that can be given to this?